engine/src/main/java/org/terasology/world/block/family/UpdatesWithNeighboursFamily.java

```diff
@@ -15,78 +15,15 @@
-import org.terasology.math.SideBitFlag;
 import org.terasology.math.geom.Vector3i;
-import org.terasology.world.BlockEntityRegistry;
-import org.terasology.world.WorldProvider;
 import org.terasology.world.block.Block;
-import org.terasology.world.block.BlockUri;
 
-import java.util.List;
-import java.util.Locale;
-
-public class UpdatesWithNeighboursFamily extends AbstractBlockFamily {
-    private ConnectionCondition connectionCondition;
-    private Block archetypeBlock;
-    private TByteObjectMap<Block> blocks;
-    private byte connectionSides;
-
-    public UpdatesWithNeighboursFamily(ConnectionCondition connectionCondition, BlockUri blockUri,
-                                       List<String> categories, Block archetypeBlock, TByteObjectMap<Block> blocks, byte connectionSides) {
-        super(blockUri, categories);
-        this.connectionCondition = connectionCondition;
-        this.archetypeBlock = archetypeBlock;
-        this.blocks = blocks;
-        this.connectionSides = connectionSides;
-
-        for (Block block : blocks.valueCollection()) {
-            block.setBlockFamily(this);
-        }
-    }
-
-    @Override
-    public Block getArchetypeBlock() {
-        return archetypeBlock;
-    }
-
-    @Override
-    public Block getBlockForPlacement(WorldProvider worldProvider, BlockEntityRegistry blockEntityRegistry, Vector3i location, Side attachmentSide, Side direction) {
-        byte connections = 0;
-        for (Side connectSide : SideBitFlag.getSides(connectionSides)) {
-            if (connectionCondition.isConnectingTo(location, connectSide, worldProvider, blockEntityRegistry)) {
-                connections += SideBitFlag.getSide(connectSide);
-            }
-        }
-        return blocks.get(connections);
-    }
-
-    public Block getBlockForNeighborUpdate(WorldProvider worldProvider, BlockEntityRegistry blockEntityRegistry, Vector3i location, Block oldBlock) {
-        byte connections = 0;
-        for (Side connectSide : SideBitFlag.getSides(connectionSides)) {
-            if (connectionCondition.isConnectingTo(location, connectSide, worldProvider, blockEntityRegistry)) {
-                connections += SideBitFlag.getSide(connectSide);
-            }
-        }
-        return blocks.get(connections);
-    }
-
-    @Override
-    public Block getBlockFor(BlockUri blockUri) {
-        if (getURI().equals(blockUri.getFamilyUri())) {
-            try {
-                byte connections = Byte.parseByte(blockUri.getIdentifier().toString().toLowerCase(Locale.ENGLISH));
-                return blocks.get(connections);
-            } catch (IllegalArgumentException e) {
-                return null;
-            }
-        }
-        return null;
-    }
-
-    @Override
-    public Iterable<Block> getBlocks() {
-        return blocks.valueCollection();
-    }
+/**
+ * Interface for Block family that gets updated by a change in a neighbor block.
+ */
+public interface UpdatesWithNeighboursFamily {
```

Comment:
Should this interface already extend `BlockFamily`. We don't have an implementation here, but I think we can be pretty sure that `UpdatesWithNeighborsFamily` will always be a `BlockFamily` as well.